pkg/scheduler/scheduler_test.go

```diff
@@ -9109,3 +9110,189 @@ func TestResourcesToReserve(t *testing.T) {
 		})
 	}
 }
+
+func TestSchedulerWithRaceCondition(t *testing.T) {
```

Comment:
```suggestion
func TestSchedulerWhenWorkloadModifiedConcurrently(t *testing.T) {
```
nit the previous name suggests the race condition is inside scheduler